scripts/build.py

```diff
@@ -0,0 +1,243 @@
+    vcpkg_executable_file_path = os.path.join(vcpkg_root_dir, 'vcpkg')
+  return vcpkg_executable_file_path
+
+
+def get_vcpkg_installation_script_path():
+  """Get absolute path to the script used to build and install vcpkg."""
+  vcpkg_root_dir = os.path.join(os.getcwd(), 'external', 'vcpkg')
+  if is_windows_os():
+    script_absolute_path = os.path.join(vcpkg_root_dir, 'bootstrap-vcpkg.bat')
+  elif is_linux_os() or is_mac_os():
+    script_absolute_path = os.path.join(vcpkg_root_dir, 'bootstrap-vcpkg.sh')
+
+  return script_absolute_path
+
+
+
+def install_cpp_dependencies_with_vcpkg(arch):
+  """Install packages with vcpkg.
+
+  This does the following,
+    - initialize/update the vcpkg git submodule
+    - build vcpkg executable
+    - install packages via vcpkg.
+  Args:
+    arch (str): Architecture (eg: 'x86', 'x64').
+  """
+  # vcpkg is a submodule of the main cpp repo
+  # Ensure that the submodule is initialized and updated
+  run_command(['git', 'submodule', 'init'])
+  run_command(['git', 'submodule', 'update'])
+
+  # Install vcpkg executable if its not installed already
+  vcpkg_executable_file_path = get_vcpkg_executable_file_path()
+  found_vcpkg_executable = os.path.exists(vcpkg_executable_file_path)
+  if not found_vcpkg_executable:
+    script_absolute_path = get_vcpkg_installation_script_path()
+    # mac example: ./external/vcpkg/bootstrap-sh
+    run_command([script_absolute_path])
+
+  # for each desktop platform, there is an existing vcpkg response file 
+  # in the repo (external/vcpkg_<triplet>_response_file.txt) defined for each target triplet
+  vcpkg_triplet = get_vcpkg_triplet(arch)
+  vcpkg_response_file_path = os.path.join(os.getcwd(), 'external', 
+                      'vcpkg_' + vcpkg_triplet + '_response_file.txt')
+
+  # mac example: ./external/vcpkg/vcpkg install @external/vcpkg_x64-osx_response_file.txt --disable-metrics
+  run_command([vcpkg_executable_file_path, 'install', '@' + vcpkg_response_file_path, '--disable-metrics'])
+
+
+def cmake_configure(build_dir, arch, build_tests=True, config=None):
+  """ CMake configure.
+
+  If you are seeing problems when running this multiple times,
+  make sure to clean/delete previous build directory.
+
+  Args:
+   build_dir (str): Output build directory.
+   arch (str): Platform Architecture (example: 'x64').
+   build_tests (bool): Build cpp unit tests.
+   config (str): Release/Debug config.
+          If its not specified, cmake's default is used (most likely Debug).
+  """   
+  cmd = ['cmake', '-S', '.', '-B', build_dir]
+  
+  # If generator is not specifed, default for platform is used by cmake, else 
+  # use the specified value
+  if config:
+    cmd.append('-DCMAKE_BUILD_TYPE={0}'.format(config))
+  if build_tests:
+    cmd.append('-DFIREBASE_CPP_BUILD_TESTS=ON')
+    cmd.append('-DFIREBASE_FORCE_FAKE_SECURE_STORAGE=ON')
+
+  vcpkg_toolchain_file_path = os.path.join(os.getcwd(), 'external', 'vcpkg', 'scripts', 'buildsystems', 'vcpkg.cmake')
```

Comment:
Do we care about sticking to the 80 character line width limit prescribed by the style guide? There are quite a few lines that exceed it. 

Though I'm not going to lie, having longer lines would be awfully convenient.

Reply:
I hate to say it, but let's try to stick with the style guide. At least the [public one](https://google.github.io/styleguide/pyguide.html). It has a link to a python formatter (yapf) that we should use.

We should probably put auto code formatting hooks into the git repo.